src/metabase/models/pulse.clj

```diff
@@ -287,6 +287,9 @@
   (let [channel (when new-channel (assoc new-channel
                                     :pulse_id       (u/get-id notification-or-id)
                                     :id             (:id existing-channel)
+                                    :enabled        (if (some? (:enabled new-channel))
+                                                      (:enabled new-channel)
+                                                      true)
```

Comment:
I don't feel awesome about this, but due to DB constraints `nil` (which is what we were always providing beforehand) defaults to true:

```

metabase=# \d pulse_channel
                                         Table "public.pulse_channel"
     Column     |           Type           | Collation | Nullable |                  Default                  
----------------+--------------------------+-----------+----------+-------------------------------------------
 enabled        | boolean                  |           | not null | true
```

My preference is to be explicit about nils in the app layer (and that lets us use the `(boolean? enabled)` constraint later on, but if you want to rely on the DB and change the schema check to `(or (nil? enabled) (boolean? enabled))` I'll hear you out.

Reply:
The `if` could be replaced by `(or (nil? (:enabled new-channel)) (:enabled new-channel))` but I found that a little hard to reason about

Reply:
so looking at that table, `enabled` is a `NOT NULL` column. I wonder where we decided that `nil` on that column should equal `true`. This would benefit from a schema defined in the code.

Reply:
Talked to Cam offline; gonna try to remove this logic and just make all the callers behave by passing in a non-nil `:enabled` value. As far as we know this is only triggered by one place in the UI that already behaves well, so I suspect it's just tests that need attention.